lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1229,6 +1232,7 @@ private TaskImpl terminateCancelCommonActions(String taskId, TaskState targetSta
     task.setModified(now);
     task.setCompleted(now);
     task.setState(targetState);
+    task = (TaskImpl) taskEndstatePreprocessorManager.processTaskBeforeEndstate(task);
```

Comment:
Why did you decide to one time put the SPI processing inside the code that modifies the task (terminateCancelCommonActions) and one time when it doesn't (after completeActionsOnTask)

I understand that otherwise you would have to put the code twice (cancel and terminate), but shouldn't we choose a consistent approach/place? What do you think?

Reply:
In all methods, I put the SPI processing right before the mapper does the updating. The update happens in "terminateCancelCommonActions" but not in  "completeActionsOnTask". That's why the processing is placed differently. We would need to refactor terminateCancelCommonActions or completeActionsOnTask to make it consistent. What do you think about that? :)

Reply:
Yes we would have to do one of these 2 options, that's what i was trying to imply with a consistent approach. Let's talk about this, I have a favourite but would like to know what you think :)